### PR TITLE
Span return `this` to enable chain calls

### PR DESCRIPTION
**Pull request by dengliming**
resolves [https://github.com/open-telemetry/opentelemetry-java/issues/1864](https://github.com/open-telemetry/opentelemetry-java/issues/1864)

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]
Thanks a lot, great change!

**Review by thisthat — APPROVED** [review]

**Review by Oberon00 — APPROVED** [review]
That was fast!

As a follow-up after this PR we could even have both span and spanbuilder implement an interface SpanAttributeSettable which has these setAttribute overloads.

File: api/src/main/java/io/opentelemetry/trace/Span.java
Comment (carlosalberto) [inline api/src/main/java/io/opentelemetry/trace/Span.java:304]: Maybe I'm kind of obsessive - but I don't want to encourage users to do something with a `Span` after `end()` ;)

**Review by carlosalberto — APPROVED** [review]

File: api/src/main/java/io/opentelemetry/trace/Span.java
Comment (Oberon00) [reply api/src/main/java/io/opentelemetry/trace/Span.java:304]: Oh, that's true! Please keep the `void` here, @dengliming

File: api/src/main/java/io/opentelemetry/trace/Span.java
Comment (Oberon00) [inline api/src/main/java/io/opentelemetry/trace/Span.java:318]: Same here (https://github.com/open-telemetry/opentelemetry-java/pull/1869/files#r510874028)

**Review by arminru — APPROVED** [review]
LGTM once `End` "returns" `void` again 🙂 Thanks!

File: api/src/main/java/io/opentelemetry/trace/PropagatedSpan.java
Comment (jkwatson) [inline api/src/main/java/io/opentelemetry/trace/PropagatedSpan.java:104]: blocking the PR until the `end` methods return void again.

**Review by jkwatson — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/opentelemetry/trace/PropagatedSpan.java
Comment (Oberon00) [reply api/src/main/java/io/opentelemetry/trace/PropagatedSpan.java:104]: both of them 😃

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/trace/PropagatedSpan.java:104]: yes. didn't feel the need to beat the dead horse any more. :)

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**
